### PR TITLE
[flutter_markdown] Soft wrapping in blockquotes

### PR DESCRIPTION
**Pull request by andrechalella**
Makes `flutter_markdown` treat paragraphs in blockquotes like a normal paragraph with respect to soft wrapping.

Currently, line breaks in blockquotes translate directly to line breaks in the output. This happens regardless of the Markdown Specification chosen. Such behavior is different from most Markdown implementations, including the reference ones cited in [flutter_markdown ](https://pub.dev/packages/flutter_markdown) ([Dart Markdown Live Editor](https://dart-lang.github.io/markdown/) and the [Markdown Live Preview](https://markdownlivepreview.com/)).

Example:

```
> my
> blockquote
```

Currently renders like:

> my
> blockquote

And this PR fixes it to become:

> my blockquote

Note: previewing in this GitHub editor, I get the hard-wrap behavior. However, [Dart Markdown Live Editor](https://dart-lang.github.io/markdown/) gives the soft-wrap behavior in all flavors, even when "GitHub Flavored Markdown" is selected. Reading the GFM spec [there's an example](https://github.github.com/gfm/#example-222) that to me implies that soft-wrapping is the correct thing in GFM, plus that's what I gather from reading [6.12](https://github.github.com/gfm/#hard-line-breaks) and [6.13](https://github.github.com/gfm/#soft-line-breaks). However, if for some reason we should hard-wrap in GFM, it should be just a matter of checking the chosen `extensionSet` when deciding whether to call `trimText()`. Seems clear to me hard-wrapping shouldn't ever be the only behavior.

Issues: this fixes https://github.com/flutter/flutter/issues/156554

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`). -- *nothing to update or add I guess*
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing. -- *couldn't get some more obscure tests to pass, but don't think it's my PR's fault*

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/flutter_markdown/pubspec.yaml
Comment (nate-thegrate) [inline packages/flutter_markdown/pubspec.yaml:7]: [Linux repo_checks](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20repo_checks/11948/overview) got mad… I think (at least one of) the issues can be fixed by changing the version number like so:

```suggestion
version: 0.7.4
```

File: packages/flutter_markdown/CHANGELOG.md
Comment (nate-thegrate) [inline packages/flutter_markdown/CHANGELOG.md:1]:
```suggestion
## 0.7.4
```

(to match pubspec)

File: packages/flutter_markdown/test/blockquote_test.dart
Comment (nate-thegrate) [inline packages/flutter_markdown/test/blockquote_test.dart:28]: I believe the only other issue is how this line has 4 empty spaces:

```suggestion

    testWidgets(
```

**Review by nate-thegrate — APPROVED** [review]
LGTM once we get the checks passing.

Thanks for the fix!

**Comment by nate-thegrate** [comment]
> previewing in this GitHub editor, I get the hard-wrap behavior.

Yeah, I've noticed the GitHub website is a bit funky, since it notices all line breaks in issue or PR comments, but if you upload a `.md` file then it follows traditional markdown rules (i.e. only break if there are 2 spaces or a `\` at the end).

**Comment by andrechalella** [comment]
Ok, it's passing now. I don't know why the version number was bad, I tried to follow the latest changes which just fixed minor stuff.

**Comment by nate-thegrate** [comment]
> I don't know why the version number was bad

I don't fully understand it either, but thanks for fixing it!

One small thing to mention—it looks like a `settings.json` file may have been added by accident.

**Comment by andrechalella** [comment]
Sorry, that was just some GitHub VS Code garbage! I think it's fixed now.

**Comment by nate-thegrate** [comment]
Awesome!

We'll be able to merge once the PR has 2 approvals.

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**
